Makefile

```diff
@@ -59,100 +59,96 @@ help:
 	@echo "------------------------------------"
 
 $(pre_deps_tag):
-	@echo "==Installing pip-tools and black=="
-	${PYTHON} -m pip install --upgrade --quiet pip
-	grep "^pip-tools\|^black"  requirements/requirements_dev.in | xargs ${PYTHON} -m pip install --quiet
-	grep "^tomli\|^setuptools"  requirements/requirements.in | xargs ${PYTHON} -m pip install --upgrade --quiet
+ifndef POETRY_EXISTS
+	@echo "Installing Poetry"
+	curl -sSL https://install.python-poetry.org | python3 -
+	which poetry
+	poetry --version
 	touch $(pre_deps_tag)
+else
+	@echo "Poetry already installed"
+	poetry --version
+	touch $(pre_deps_tag)
+endif
 
-requirements/requirements.txt: requirements/requirements_dev.txt
-	@echo "==Compiling requirements.txt=="
-	cat requirements/requirements.in > subset.in
-	echo "" >> subset.in
-	echo "-c requirements/requirements_dev.txt" >> subset.in
-	pip-compile --output-file "requirements/requirements.txt" --quiet --no-emit-index-url subset.in
-	rm subset.in
-
-reqs: requirements/requirements.txt
-
-requirements/requirements_dev.txt: requirements/requirements_dev.in requirements/requirements.in $(pre_deps_tag)
-	@echo "==Compiling requirements_dev.txt=="
-	pip-compile --output-file requirements/requirements_dev.txt --quiet --no-emit-index-url requirements/requirements.in requirements/requirements_dev.in
+$(setup_tag): $(pre_deps_tag) pyproject.toml
+	@echo "==Setting up package environment=="
+	poetry config virtualenvs.prefer-active-python true
+	poetry lock --no-update
+	poetry install --with unit --no-cache
+	touch $(setup_tag)
 
-reqs_dev: requirements/requirements_dev.txt
+requirements/requirements.txt: poetry.lock pyproject.toml
+	mkdir -p requirements
+	poetry export -f requirements.txt --output requirements/requirements.txt --without dev --without-hashes
 
-$(env_tag): requirements/requirements.txt
-	@echo "==Installing requirements.txt=="
-	pip-sync --quiet requirements/requirements.txt
-	rm -rf $(install_tag)
-	touch $(env_tag)
+reqs: requirements/requirements.txt
 
-$(env_dev_tag): requirements/requirements_dev.txt
-	@echo "==Installing requirements_dev.txt=="
-	pip-sync --quiet requirements/requirements_dev.txt
-	rm -rf $(install_tag)
-	touch $(env_dev_tag)
+$(env_tag): $(pre_deps_tag) pyproject.toml
+	@echo "==Setting up package environment=="
+	poetry config virtualenvs.prefer-active-python true
+	poetry lock --no-update
+	poetry install --no-cache
+	touch $(setup_tag)
 
 setup: $(env_tag)
-	@echo "==Setting up package environment=="
-	rm -f $(env_dev_tag)
 
-setup_dev: $(env_dev_tag)
-	@echo "==Setting up development environment=="
-	rm -f $(env_tag)
+setup_dev: $(env_tag)
+	poetry install --with dev --sync --no-cache
```

Comment:
Since setup and setup_dev are based on the same tag how could one switch from a "dev environment" (i.e. an environment where dev requirements are installed) to a "regular environment" (i.e. an environment where only package requirements are installed)?
I mean using make commands, of course...

Reply:
tested and it is working as expected. Doing `make setup` installs direct dependencies, `make setup_dev` installs also dev dependencies.